src/analysis/rule_extraction.jl

```diff
@@ -77,16 +77,97 @@ end
 """
     print_rules(rules::Vector{Rule})
 
-Print all rules in a human readable way. SIRUS also has this functionality but 1) it only 
-prints once, when the model is fitted and 2) this function prints the rules in a more 
+Print all rules in a human readable way. SIRUS also has this functionality but 1) it only
+prints once, when the model is fitted and 2) this function prints the rules in a more
 compact form.
 
 # Arguments
 - `rules` : Vector of Rule objects
 """
-function print_rules(rules::Vector{Rule{Vector{Float64},Vector{Vector{Any}}}})
+function print_rules(rules::Vector{Rule{Vector{Vector},Vector{Float64}}})
     condition(rule) = [c == :L ? "$(c[1]) < $(c[3])" : "$(c[1]) ≥ $(c[3])" for c in rule.condition]
     consequent(rule) = " then $(rule.consequent[1]) else $(rule.consequent[2])\n"
     rule_string(rule) = "if " * join(condition(rule), " ") * consequent(rule)
     print(join([rule_string(rule) for rule in rules]))
 end
+
+
+"""
+    cluster_rules(clusters::Vector{T}, X::DataFrame, outcomes::AbstractMatrix{F}, max_rules::Int64; n_trees::Int64=1000) where {T<:Integer,F<:Real}
+
+Use SIRUS package to extract rules from time series clusters based on some summary metric (default is median)
+
+# Arguments
+- `clusters` - Vector of cluster indexes for each scenario outcome
+- `X::DataFrame` - DataFrame with factors to be used as input by SIRUS
+- `outcomes` - Matrix of some metric over time for all scenarios
+- `max_rules` - Maximum number of rules, to be used as input by SIRUS
+- `n_trees` - Number of trees to be created by SIRUS algorithm
```

Comment:
Julia best practice suggests not to repeat information that is provided in the function signature (e.g., the types)

Note: I've opted to replace `factors` with `features` because that's the usual terminology for Random Forest/ML - feel free to change this back to `factors` to be consistent within this documentation.

```suggestion
- `clusters` : Cluster indexes for each scenario outcome
- `X` : Features to be used as input for SIRUS
- `outcomes` : Matrix of some metric over time for all scenarios
- `max_rules` : Maximum number of rules, to be passed to SIRUS
- `n_trees` : Number of trees to be created by SIRUS
```